example/App.js

```diff
@@ -4,46 +4,97 @@
  * @flow
  */
 
-import React, { Component } from 'react';
+import React, { Component } from 'react'
+import { Platform, StyleSheet, Text, View, Button, Picker } from 'react-native'
 import {
-  Platform,
-  StyleSheet,
-  Text,
-  View
-} from 'react-native';
+  Emoji,
+  NimblePicker,
+  EmojiButton,
+  ModalPicker,
+} from 'emoji-mart-native'
+import data from 'emoji-mart-native/data/all.json'
+import dataRequires from 'emoji-mart-native/data/local-images/all'
+import { SetPicker } from './components'
 
-const instructions = Platform.select({
-  ios: 'Press Cmd+R to reload,\n' +
-    'Cmd+D or shake for dev menu',
-  android: 'Double tap R on your keyboard to reload,\n' +
-    'Shake or press menu button for dev menu',
-});
+const { emojis: localEmojis } = dataRequires
+export default class App extends Component {
+  state = {
+    set: 'twitter',
+    selectedEmoji: { id: 'santa' },
+    showEmojiPicker: false,
+  }
+  emojiSelectTrigger = (emoji) => {
+    this.setState({ selectedEmoji: emoji })
+  }
 
-type Props = {};
-export default class App extends Component<Props> {
+  showPickerTrigger = (visible) => {
+    this.setState({ modalVisible: visible })
+  }
   render() {
     return (
       <View style={styles.container}>
-        <Text style={styles.welcome}>
-          Welcome to React Native!
-        </Text>
-        <Text style={styles.instructions}>
-          To get started, edit App.js
-        </Text>
-        <Text style={styles.instructions}>
-          {instructions}
-        </Text>
+        <View style={styles.titleContainer}>
+          <Text style={styles.titleText}>Emoji Mart Native</Text>
+          <Emoji emoji="department_store" size={16} />
+        </View>
+        <SetPicker
+          selectedValue={this.state.set}
+          onValueChange={(itemValue, itemIndex) =>
+            this.setState({
+              set: itemValue,
+              showEmojiPicker: true,
+            })
+          }
+        />
+        <ModalPicker
+          isVisible={this.state.modalVisible}
+          showCloseButton={true}
+          onPressClose={() => {
+            this.showPickerTrigger(false)
+          }}
+          set={this.state.set}
+          data={data}
+          onSelect={this.emojiSelectTrigger}
```

Comment:
@ex7r3me we should also close the modal picker after selecting an emoji with it

Reply:
Good point, now it closes on select trigger :+1: